FWIO/RNTuple/plugins/RNTupleOutputFile.cc

```diff
@@ -0,0 +1,583 @@
+      case CompressionAlgos::kZSTD:
+        return RCompressionSetting::EAlgorithm::kZSTD;
+      case CompressionAlgos::kZLIB:
+        return RCompressionSetting::EAlgorithm::kZLIB;
+      case CompressionAlgos::kLZ4:
+        return RCompressionSetting::EAlgorithm::kLZ4;
+    }
+    return RCompressionSetting::EAlgorithm::kZSTD;
+  }
+}  // namespace
+
+using namespace ROOT::Experimental;
+namespace edm {
+  RNTupleOutputFile::RNTupleOutputFile(std::string const& iFileName,
+                                       FileBlock const& iFileBlock,
+                                       SelectedProductsForBranchType const& iSelected,
+                                       Config const& iConfig,
+                                       bool anyProductProduced)
+      : file_(iFileName.c_str(), "recreate", ""),
+        wrapperBaseTClass_(TClass::GetClass("edm::WrapperBase")),
+        selectorConfig_(iConfig.selectorConfig),
+        dropMetaData_(iConfig.dropMetaData) {
+    setupRuns(iSelected[InRun], iConfig);
+    setupLumis(iSelected[InLumi], iConfig);
+    setupEvents(iSelected[InEvent], iConfig, anyProductProduced);
+    setupPSets(iConfig);
+    setupParentage(iConfig);
+    setupMetaData(iConfig);
+
+    auto const& branchToChildMap = iFileBlock.productDependencies().childLookup();
+    for (auto const& parentToChildren : branchToChildMap) {
+      for (auto const& child : parentToChildren.second) {
+        productDependencies_.insertChild(parentToChildren.first, child);
+      }
+    }
+  }
+
+  namespace {
+    std::string fixBranchName(std::string const& iName) {
+      //need to remove the '.' at the end of the branch name
+      return iName.substr(0, iName.size() - 1);
+    }
+
+    void noSplitField(ROOT::Experimental::RFieldBase& iField) {
+      auto const& typeName = iField.GetTypeName();
+      if (typeName == "std::uint16_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt16}});
+      } else if (typeName == "std::uint32_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt32}});
+      } else if (typeName == "std::uint64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt64}});
+      } else if (typeName == "std::int16_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt16}});
+      } else if (typeName == "std::int32_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt32}});
+      } else if (typeName == "std::int64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt64}});
+      } else if (typeName == "float") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal32}});
+      } else if (typeName == "double") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal64}});
+      }
+    }
+
+    void checkSubFieldsForNoSplit(ROOT::Experimental::RFieldBase& iField,
```

Comment:
With the `check` prefix, I'd think this function would only inspect the subfields (and then e.g. return something or throw an exception). How about changing the name to something that indicates better the subfields being changed to no-split?

Reply:
name changed to `findSubFieldsForNoSplitThenApply`